File: Apps/Backend/TaskService/src/main/java/com/demo/TaskService/DTOs/Responses/ServiceResult.java
package com.demo.TaskService.DTOs.Responses;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ServiceResult<T> {
	boolean isSuccess;
	String message;
	T data;

	public static <T> ServiceResult<T> success(T data, String message) {
		return new ServiceResult<>(true, message, data);
	}

	public static <T> ServiceResult<T> failure(String message) {
		return new ServiceResult<>(false, message, null);
	}
}

File: Apps/Backend/IdentityService/DTOs/ServiceResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace IdentityService.DTOs
{
    public class ServiceResult<T>
    {
        public bool IsSuccess { get; set; } = false;
        public string? Message { get; set; }
        public T? Data { get; set; }

        public static ServiceResult<T> Ok(T data, string? message = null)
            => new ServiceResult<T> { IsSuccess = true, Data = data, Message = message };

        public static ServiceResult<T> Fail(string message)
            => new ServiceResult<T> { IsSuccess = false, Message = message };

    }
}